### DISCUSSION TITLE
How can I get the value of decision varible during optimization? But NOT after optimization.

### DISCUSSION THREAD

**Discussion by Vonct** [casadi/casadi#3317] [Q&A]
Hi, currently I have a problem about how to convert a `MX` to a `double` .
   The code is like:
```
X = opti.variable(4,N_+1);
for(int i = 1; i < N_; i++){
     double s_next = static_cast<double>(X(3,i)); //this will raise an error
    calc_position(s_next); //calc_position function needs a double 
} 
```
   I want to get the value of the decision variable, so that I can use it in a calculation. 
   I wonder if my idea can be impleted.

**Comment by jaeandersson** [comment]
Please revisit the user guide to make sure you understand how CasADi expressions work. To get a numerical value from an `MX` expression, you typically have to use the expression to define a function, then evaluate the function.

**Comment by jgillis** [comment]
If you have computations that are not expressible using CasADi symbols, you can out those computations in a Callback https://web.casadi.org/docs/#sec-callback
